### PR TITLE
Copter: mode_auto: retract the landing gear automatically

### PR DESCRIPTION
**Pull request by liang-tang**
retract the landing gear automatically when takeoff finish.

### CODE REVIEW

**Comment by peterbarker** [comment]
Perhaps we should add a mission item instead so people have the option of retracting or not?

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -1490,7 +1490,14 @@ void Copter::ModeAuto::do_RTL(void)
 bool Copter::ModeAuto::verify_takeoff()
 {
     // have we reached our target altitude?
-    return copter.wp_nav->reached_wp_destination();
+    bool reached_wp_dest = copter.wp_nav->reached_wp_destination();
```
Comment (rmackay9) [inline ArduCopter/mode_auto.cpp]: this can be a const.

File: ArduCopter/mode_auto.cpp [hunk 2]
```diff
@@ -1490,7 +1490,14 @@ void Copter::ModeAuto::do_RTL(void)
...
+
+    // retract the landing gear
+    if (SRV_Channels::function_assigned(SRV_Channel::k_landing_gear_control) && reached_wp_dest) {
```
Comment (rmackay9) [inline ArduCopter/mode_auto.cpp]: We shouldn't make the assumption that the landing gear is triggered by an rc output channel.  We should just assuming the landing gear function will ignore the command if no output channel is defined.

**Comment by rmackay9** [comment]
i agree with Peter's comments but I'm inclined to merge this (after the few changes I've mentioned above.  The reason is that we already auto-deploy the gear (we don't require a mission item) so I think it makes sense to auto-retract as well.  I don't think users will complain..

File: ArduCopter/mode_auto.cpp (hunk 2 above)
Comment (liang-tang) [reply ArduCopter/mode_auto.cpp]: > We shouldn't make the assumption that the landing gear is triggered by an rc output channel. We should just assuming the landing gear function will ignore the command if no output channel is defined.

How should I update this part?

Comment (rmackay9) [reply ArduCopter/mode_auto.cpp]: @liang-tang, just remove the "function_assigned" check completely.

Comment (liang-tang) [reply ArduCopter/mode_auto.cpp]: like this "if (reached_wp_dest) { "?

Comment (rmackay9) [reply ArduCopter/mode_auto.cpp]: @liang-tang, yes, that sounds right

Comment (liang-tang) [reply ArduCopter/mode_auto.cpp]: done!

**Comment by khancyr** [comment]
changes were made and this is good to merge

**Comment by tridge** [comment]
it would be nice to add a LGR_OPTIONS bitmask with a bit for "retract on takeoff"

**Comment by peterbarker** [comment]
```
UTC0040 - Copter mode auto retract landing gear automatically
Maybe use a parameter to specify behaviour
We have a series of landing gear parameters already
LGR_OPTIONS bitmask
```

**Pull request merged**